packages/debridge/src/hooks/useDebridge.tsx

```diff
@@ -0,0 +1,92 @@
+import { IStore } from '@gardenfi/utils';
+import React, { ReactNode, useContext, useEffect, useState } from 'react';
+import { Debridge } from '../debridge';
+import { WalletClient } from 'viem';
+import { DEBRIDGE_TXS_CACHE_KEY } from 'src/constants';
+import { DeBridgeTransaction } from 'src/debridge.api.types';
+import { SwapConfig, SwapResponse } from 'src/debridge.types';
+import { AsyncResult, Err, Ok, Result } from '@catalogfi/utils';
+//we don't have to do all these imports just the chains we are going to support
+
+type DebridgeContextType = {
+  debridge: Debridge;
+  store: IStore;
+  swap: (swapConfig: SwapConfig) => AsyncResult<SwapResponse, string>;
+  txs: DeBridgeTransaction[];
+};
+
+const defaultDebridgeValue = {
+  debridge: new Debridge({
+    debridgeDomain: 'https://api.dln.trade/v1.0',
+    debridgeTxDomain: 'https://stats-api.dln.trade/api/Orders',
+    debridgePointsDomain: 'https://points-api.debridge.finance/api/points',
+  }),
+  store: localStorage,
+  swap: async (swapConfig: SwapConfig): AsyncResult<SwapResponse, string> =>
+    Ok({} as SwapResponse),
+  txs: [],
+};
+
+const DebridgeContext =
+  React.createContext<DebridgeContextType>(defaultDebridgeValue);
+
+export const useDebridge = () => useContext(DebridgeContext);
+
+export type UseDebridgeProps = {
+  debridge: Debridge;
+  store: IStore;
+  client: WalletClient;
+};
+
+export const DebridgeProvider = ({
+  children,
+  address,
+}: {
+  children: ReactNode;
+  address: `0x${string}`;
+}) => {
+  const [txs, setTxs] = useState<DeBridgeTransaction[]>([]);
+  const cacheKey = DEBRIDGE_TXS_CACHE_KEY + address.toLowerCase();
+  const debridge = defaultDebridgeValue.debridge;
+
+  useEffect(() => {
+    (async () => {
+      const debridgeTxs = await debridge.getTxs({
+        address,
+      });
+
+      defaultDebridgeValue.store.setItem(
+        cacheKey,
+        JSON.stringify(debridgeTxs.val.orders)
+      );
+
+      if (debridgeTxs.error) return;
+      setTxs(debridgeTxs.val.orders);
+    })();
+  }, [address]); //txs change with address
+
+  const swap = async (
+    swapConfig: SwapConfig
+  ): AsyncResult<SwapResponse, string> => {
+    const swapRes = await debridge.swap(swapConfig);
+
+    if (swapRes.error) return Err(swapRes.error);
+
+    const txHash = swapRes.val.txHash;
+    const tx = await debridge.getTx(txHash);
+
+    if (tx.error) return Err(tx.error);
+
+    setTxs((txs) => [...txs, tx.val.orders[0]]);
```

Comment:
write a wrapper for setting the txns state which sets the tx state and localstorage